test/browser/lifecycle.js

```diff
@@ -1325,6 +1325,38 @@ describe('Lifecycle methods', () => {
 	});
 
 
+	describe('#setState', () => {
```

Comment:
Could I suggest also including a test of the function parameter version of the function.  It looks like you are passing `prev` by reference here, so wouldn't any changes to it also mutate prev?  It would be interesting to explore the effects of that.

Reply:
Good suggestion, I just feel like a full `setState` test suite is somewhat out of scope for the PR. Strange that it's missing though.

Yes it passes the previous state by reference, exactly as it should. Treating the previous state as immutable is the responsibility of the user.